### PR TITLE
Fix #341: Support typed parameters for the On_propertyName_Changed method

### LINKED ISSUE #341: Have On_property_Changed events also deliver the old value with special signature
The `On_property_Changed` is very useful when adding property changed watchers when a property has changed.

Unfortunately the same mechanism cannot be used to unregister the previous delegate attached to the previous object.

It would be nice if we could either extend `On_property_Changed` so that when given with a signature such as:

   void OnNameChanged(string oldValue)

it would provide the old value as well so we could undo whatever was done to that property before.

Probably it would need to be a nullable to also support value types.

Example:

```
class A {
   public SomeOtherClass Classy { get; set;}

   private OnClassyChanged(SomeOtherClass old){
      if (old != null) {
          old.PropertyChanged -= WatchChanges;
      }

      if (Classy != null) {
          Classy.PropertyChanged += WatchChanges;
      }
}
```

### PR DESCRIPTION
**Pull request by tom-englert**

### CODE REVIEW

**Comment by tom-englert** [comment]
Wiki is already prepared

File: Tests/AssemblyToProcessTests.cs [hunk 1]
```diff
@@ -330,6 +330,81 @@ public void OnPropertyNameChangedMethodWithBeforeAfterIsCalled()
...
+        var instance = testResult.GetInstance(nameof(ClassWithOnChangedBeforeAfterTypedInvalidSignatureDefault));
+        instance.Property1 = "foo";
+
+        Assert.Null(instance.OnProperty1ChangedCalled);
+
+        instance.Property1 = "bar";
+
+        Assert.Null(instance.OnProperty1ChangedCalled);
+
+        Assert.DoesNotContain(testResult.Warnings, w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedBeforeAfterTypedInvalidSignatureDefault)));
```
Comment (ltrzesniewski) [inline Tests/AssemblyToProcessTests.cs:364]: Don't we want a warning to be emitted in this case?

Comment (tom-englert) [reply Tests/AssemblyToProcessTests.cs]: Yes, missed that.

**Review by ltrzesniewski — APPROVED** [review]
Looks good :+1:

**Pull request merged**